regexp-lexer.js

```diff
@@ -306,7 +306,10 @@ RegExpLexer.prototype = {
             }
         }
 
-        lines = match[0].match(/(?:\r\n?|\n).*/g);
+        // Bail from match if string doesn't contain newline nor carriage return
+        if (lines.indexOf('\n') != -1 || lines.indexOf('\r') != -1) {
```

Comment:
nitpick: `lines` isn't initialized here yet. `indexOf()` should be applied against `match[0]` instead.